scanner.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/schollz/progressbar/v3"
+	"github.com/sirupsen/logrus"
+)
+
+func CloneRepo(cloneURL string) (string, error) {
+	dir, err := os.MkdirTemp("", "repo-*")
+	if err != nil {
+		return "", err
+	}
+
+	cmd := exec.Command("git", "clone", "--depth=1", cloneURL, dir)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	logrus.Infof("⏳ Cloning repository into %s", dir)
+	err = cmd.Run()
+	if err != nil {
+		return "", err
+	}
```

Comment:
When a function returns a single output and this output is checked in an if (and only in this if) it should be one line:
```
if err = cmd.Run(); err != nil {
	return "", err
}
```

This is a common pattern in error handling